crates/nu-command/src/debug/info.rs

```diff
@@ -34,25 +47,186 @@ impl Command for DebugInfo {
             .category(Category::Debug)
     }
 
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        _call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let span = Span::unknown();
+
+        let record = LazySystemInfoRecord { span };
+
+        Ok(Value::lazy_record(Box::new(record), span).into_pipeline_data())
+    }
+
     fn examples(&self) -> Vec<Example> {
         vec![Example {
             description: "View process information",
             example: "debug info",
             result: None,
         }]
     }
+}
 
-    fn run(
+#[derive(Debug, Clone)]
+struct LazySystemInfoRecord {
+    span: Span,
+}
+
+impl LazySystemInfoRecord {
+    fn get_column_value_with_system(
         &self,
-        _engine_state: &EngineState,
-        _stack: &mut Stack,
-        _call: &Call,
-        _input: PipelineData,
-    ) -> Result<PipelineData, ShellError> {
-        let span = Span::unknown();
-        // get the nushell process id
+        column: &str,
+        system_option: Option<&System>,
+    ) -> Result<Value, ShellError> {
         let pid = Pid::from(std::process::id() as usize);
-        // only refresh the process and memory information
+        match column {
+            "threadid" => Ok(Value::int(get_thread_id(), self.span)),
```

Comment:
I'm getting `8153895680` as a thread_id on MacOS. Is that reasonable? I don't know, just seemed weird to me.
```suggestion
            "thread_id" => Ok(Value::int(get_thread_id(), self.span)),
```

Reply:
This should be correct AFAIK, phthread_self returns the current thread id in libc: https://opensource.apple.com/source/Libc/Libc-583/pthreads/pthread_self.3.auto.html

Reply:
Thanks. Just looked weird to have such a large number as a thread_id.

Reply:
Yeah AFAIK this isn't a monotonously increasing number, possibly just returns something like the pointer to the tread handle.